### PR TITLE
SDLC: Add overview, GH issues as SSOT

### PR DESCRIPTION
**Pull request by jhchabran**
Follow-up to [RFC 686](https://docs.google.com/document/d/1p5_FheDnShXJ1P3gt2LWUc4tqU4zRvECnG7SZhzB398/edit#).

- Removes mention to other trackers.
- Promotes GitHub issues at the single source of truth.
- Add a paragraph about teams detailing further the process.
- Add a how to contribute section, making it explicit that this is a high friction page, that shouldn't be updated often and without great care, as it affects compliance.

### CODE REVIEW

**Comment by jhchabran** [comment]
@taylorsperry, @quinnhare and @kstretch9, I've updated this page, with the main changes listed in the PR's description. Could you take a look before we submit it for review on #wg-soc2 ? 

The workflow itself, is still blurry and needs to be refined, but that's outside the goal of the present effort.

File: content/departments/engineering/process/sdlc.md [hunk 1]
```diff
@@ -1,35 +1,78 @@
...
+### Roadmap items
+
+At the very top, Sourcegraph uses roadmap items, found in the [product and engineering roadmap tracker](https://github.com/sourcegraph/product-engineering-tracker) list objectives for each quarter in individual issues.
+Each individual issue is tagged with their owning organization, owning team and assignees, who are in charge of updating the issue during the quarter.
+
+The issue content contains the following sections:
+
+- Problem definition
+- Measure of success
+- Solution summar
```
Comment (marekweb) [inline content/departments/engineering/process/sdlc.md]:
```suggestion
- Solution summary
```

File: content/departments/engineering/process/sdlc.md [hunk 2]
```diff
@@ -1,35 +1,78 @@
 # Software development life cycle
 
-<span class="badge badge-note">SOC2/GN-98</span>
+<span class="badge badge-note">SOC2/CI-98</span>
 
-Sourcegraph uses three types of approaches to drive changes:
+_Please follow this [guidance](#contributing-to-this-page) on how to contribute to this page._
```
Comment (bobheadxi) [inline content/departments/engineering/process/sdlc.md]:
```suggestion
> NOTE: Please follow [this guidance](#contributing-to-this-page) on how to contribute to this page.
```

File: content/departments/engineering/process/sdlc.md [hunk 3]
```diff
@@ -1,35 +1,78 @@
...
+
+This document explains the workflow for driving changes on the Sourcegraph software application.
+
+## Overview
+
+Sourcegraph uses GitHub issues to track and drive changes to its application. They are the single source of truth, and are structured as following:
+
+- Roadmap items
```
Comment (bobheadxi) [inline content/departments/engineering/process/sdlc.md]:
```suggestion
- Roadmap issues
```

File: content/departments/engineering/process/sdlc.md [hunk 4]
```diff
@@ -1,35 +1,78 @@
...
+  - Tracking issues
+    - Standard issues
+
+### Roadmap items
+
+At the very top, Sourcegraph uses roadmap items, found in the [product and engineering roadmap tracker](https://github.com/sourcegraph/product-engineering-tracker) list objectives for each quarter in individual issues.
```
Comment (bobheadxi) [inline content/departments/engineering/process/sdlc.md]:
```suggestion
At the very top, Sourcegraph uses roadmap issues, found in the [product and engineering roadmap tracker](https://github.com/sourcegraph/product-engineering-tracker) list objectives for each quarter in individual issues.
```

File: content/departments/engineering/process/sdlc.md [hunk 5]
```diff
@@ -1,35 +1,78 @@
...
+Each individual issue is tagged with their owning organization, owning team and assignees, who are in charge of updating the issue during the quarter.
+
+The issue content contains the following sections:
```
Comment (bobheadxi) [inline content/departments/engineering/process/sdlc.md]:
```suggestion
A roadmap issue contains the following sections:
```

File: content/departments/engineering/process/sdlc.md [hunk 6]
```diff
@@ -1,35 +1,78 @@
 # Software development life cycle
 
-<span class="badge badge-note">SOC2/GN-98</span>
+<span class="badge badge-note">SOC2/CI-98</span>
```
Comment (bobheadxi) [inline content/departments/engineering/process/sdlc.md:3]: I wonder if we should keep both badges for now, for ease of reference?

**Review by bobheadxi — APPROVED** [review]
LGTM, some minor nits :) Thanks for updating the control point as well!

File: content/departments/engineering/process/sdlc.md (hunk 6 above)
Comment (jhchabran) [reply content/departments/engineering/process/sdlc.md:3]: I was thinking about updating all of them as that's the new refs we're using internally. Why would we keep the old ones?

**Pull request merged**